cpp/src/generators/generator_tools.cu

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cugraph/graph_generators.hpp>
+#include <cugraph/utilities/error.hpp>
+#include <experimental/scramble.cuh>
+
+#include <raft/cuda_utils.cuh>
+
+#include <rmm/device_uvector.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/execution_policy.h>
+#include <thrust/for_each.h>
+
+#include <numeric>
+
+namespace cugraph {
+
+namespace detail {
+
+template <typename T>
+void append_all(raft::handle_t const &handle,
+                std::vector<rmm::device_uvector<T>> &&input,
+                rmm::device_uvector<T> &output)
```

Comment:
Should we better return rmm::device_uvector<T> instead of taking ```rmm::device_uvector<T>& output```. Unless the ```ouput``` is in-out or there is another strong reason to do so, returning is more functional and side-effects-free than taking an lvalue reference.

Reply:
Done in next push